### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by pmarsh-scottlogic**
- used JSR303 for validation with annotations
- Validated API new order post request
- added unit tests to make sure they worked

Didn't figure out how to use annotations to validate PathVariables so did it manually there instead. I'm pretty sore one forum said that using annotations for this isn't supported
e.g.
```   
@GetMapping(value = "/orderbook/sell/{accountId}")
    public ResponseEntity<List<OrderbookItem>> orderbook_sell(@PathVariable String accountId) {
        try {
            UUID accountUuid = UUID.fromString(accountId);
            return ResponseEntity.ok(orderService.getOrderbook(OrderAction.SELL, accountUuid));
        }
        catch(Exception e) {
            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
            // how to include error message in body when I've already specified the response body type?
        }
    }
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -15,8 +15,11 @@ repositories {
 }
 
 dependencies {
+//	implementation 'org.springframework.boot:spring-boot-starter-data-jpa'
```
Comment (nataliadombrowska) [inline build.gradle]: If you have any unused dependencies i would just delete them before merging it to main.

File: src/main/java/com/example/demo/matcher/MatcherController.java [hunk 2]
```diff
@@ -27,31 +29,30 @@ public class MatcherController {
 
     @GetMapping(value = "/orderbook/buy")
     public ResponseEntity<List<OrderbookItem>> orderbook_buy() {
-        return new ResponseEntity<>(orderService.getOrderbook(OrderAction.BUY), HttpStatus.OK);
+        return ResponseEntity.ok(orderService.getOrderbook(OrderAction.BUY));
```
Comment (nataliadombrowska) [inline src/main/java/com/example/demo/matcher/MatcherController.java:32]: Ooh nice that looks cleaner :)

File: src/main/java/com/example/demo/matcher/models/NewOrderParams.java [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
 import lombok.ToString;
 
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Pattern;
+
 @ToString @AllArgsConstructor @Getter
 public class NewOrderParams {
+    @UUID(message = "Bad UUID format")
```
Comment (nataliadombrowska) [inline src/main/java/com/example/demo/matcher/models/NewOrderParams.java:15]: Nice! Where are we anticipating to see this message?

File: src/main/java/com/example/demo/matcher/models/Order.java [hunk 4]
```diff
@@ -18,6 +18,11 @@ public class Order {
     private final OrderAction action;
     private final Long datetime;
 
+    public static final int minPrice = 0;
+    public static final int maxPrice = 1000000000;
```
Comment (nataliadombrowska) [inline src/main/java/com/example/demo/matcher/models/Order.java:22]: Where is this constant used?

**Review by nataliadombrowska — APPROVED** [review]
No blockers from me just a couple things like deleting the dependencies that are commented out and a couple comments to clear things up for me 👍 Very good job I like the way you're testing it too :)

File: src/main/java/com/example/demo/matcher/models/Order.java (hunk 4 above)
Comment (pmarsh-scottlogic) [reply src/main/java/com/example/demo/matcher/models/Order.java:22]: setting @min and @max validation constraints in the NewOrderParams class, which is responsible for receiving information about the new order at the endpoint

File: src/main/java/com/example/demo/matcher/models/NewOrderParams.java (hunk 3 above)
Comment (pmarsh-scottlogic) [reply src/main/java/com/example/demo/matcher/models/NewOrderParams.java:15]: When there is a bad UUID in a new order, the API will return a BadRequest response and the body will include this error message

File: build.gradle (hunk 1 above)
Comment (pmarsh-scottlogic) [reply build.gradle]: Oops, I did it again!

File: src/main/java/com/example/demo/matcher/models/NewOrderParams.java (hunk 3 above)
Comment (pmarsh-scottlogic) [reply src/main/java/com/example/demo/matcher/models/NewOrderParams.java:15]: Like so:
image

File: src/main/java/com/example/demo/matcher/models/Order.java (hunk 4 above)
Comment (pmarsh-scottlogic) [reply src/main/java/com/example/demo/matcher/models/Order.java:22]: image

**Pull request merged**